nucliadb_node/src/shards/shard_writer.rs

```diff
@@ -540,30 +540,56 @@ impl ShardWriter {
 
     #[tracing::instrument(skip_all)]
     pub fn merge(&self, context: MergeContext) -> NodeResult<MergeMetrics> {
-        // TODO: merge and returns metrics of all vectorsets
-        let merge_result = {
+        let mut merge_results = HashMap::new();
+
+        {
             let indexes: &ShardWriterIndexes = &read_rw_lock(&self.indexes);
-            let default_vectors_index = indexes
-                .vectors_indexes
-                .get(DEFAULT_VECTORS_INDEX_NAME)
-                .expect("Default vectors index should never be deleted (yet)");
-            let runner = default_vectors_index.prepare_merge(context.parameters)?;
-            let Some(mut runner) = runner else {
-                return Ok(MergeMetrics {
-                    merged: 0,
-                    left: 0,
-                });
-            };
-            runner.run()?
-        };
+            for (name, vectors_index) in indexes.vectors_indexes.iter() {
+                let runner = vectors_index.prepare_merge(context.parameters);
+                if let Ok(Some(mut runner)) = runner {
+                    let result = runner.run();
```

Comment:
`runner.run()` must be outside of any locks

Reply:
Fixed